blockonomics-woocommerce.php

```diff
@@ -501,10 +501,16 @@ function blockonomics_activation_hook() {
 }
 
 // Since WP 3.1 the activation function registered with register_activation_hook() is not called when a plugin is updated.
+// Adding timestamp column in db to handle underpayments
 function blockonomics_update_db_check() {
     global $wpdb;
     global $blockonomics_db_version;
 
+    $timestamp_column = $wpdb->get_results(  "SELECT COLUMN_NAME FROM INFORMATION_SCHEMA.COLUMNS
+    WHERE table_name = 'wp_blockonomics_orders' AND column_name = 'timestamp'"  );
+    if(empty($timestamp_column)){
+        $wpdb->query("ALTER TABLE wp_blockonomics_orders ADD timestamp INT DEFAULT NULL");
+    }
```

Comment:
Be very careful what you put in the blockonomics_update_db_check() function. It is loaded for every PHP page by `plugins_loaded` . Add a doc comment on top of function for this
The upgrade code must be inside  below if statement

Reply:
These upgrade changes should be moved into their own db version if statement and the database version incremented: `$blockonomics_db_version = '1.2';`
```
if ($installed_ver < 1.2) {
  // Upgrade code
}
```

Using the `plugins_loaded` hook is the recommended method for running the upgrade to ensure this runs on plugin updates. More information [here](https://codex.wordpress.org/Creating_Tables_with_Plugins#Adding_an_Upgrade_Function).